### DISCUSSION TITLE
Will dm+d codelists identify deprecated dm+d ids?

### DISCUSSION THREAD

**Discussion by brianmackenna** [opensafely/documentation#465] [Codelists]
In OpenCodelists you can create a codelist using the current version of the [NHS dictionary of medicines and devices (dm+d)](https://www.thedatalab.org/blog/81/what-is-the-dmd-the-nhs-dictionary-of-medicines-and-devices/), the NHS standard for prescribing.

My question is will this codelist identify medications where the code has been deprecated and replaced with a new code?

For example the NHS Digital term browser tells us at one point that [Floxacillin 250mg capsules VMP 323464005](https://termbrowser.nhs.uk/?perspective=full&conceptId1=404684003&edition=uk-edition&release=v20210929&server=https://termbrowser.nhs.uk/sct-browser-api/snomed&langRefset=999001261000000100,999000691000001104) has been deprecated. This would now come under [VMP 39692111000001101](https://openprescribing.net/dmd/vmp/39692111000001101/)

If I run a codelist with vmp/39692111000001101 would it pick up preparations from when VMP 323464005 was live?  (i think this might be a question for system suppliers to see if there is mapping done to deprecated VMPs/AMPs)

Two further points
1.  My first inclination was that  this specific example was not a good practical example for OpenSAFELY right now as the name changes/deprecations (i.e. the changes in international standards on antibiotic spellings) were so long ago but I'm sure there are good examples. However @inglesp has checked and VMP 323464005 was in TRUD last year. We may need to do more work to see implications depending on answer to the question above. 
2. There is a separate but related issue -if an AMP is discontinued and removed from dm+d it won't show up in OpenSAFELY code builder. I'm not sure how much of a problem this is as even a lot of discontinued products are marked and reamin in dm+d but maybe @inglesp has thought about this? I can't see an issue in [OpenCodelists repo](https://github.com/opensafely-core/opencodelists/issues)

**Comment by inglesp** [comment]
> There is a separate but related issue -if an AMP is discontinued and removed from dm+d it won't show up in OpenSAFELY code builder. I'm not sure how much of a problem this is as even a lot of discontinued products are marked and reamin in dm+d but maybe @inglesp has thought about this?

This affects BNF codes too.  Have created https://github.com/opensafely-core/opencodelists/issues/938.  The priority for addressing this will be determined by (a) whether suppliers update data to account for deprecated products and (b) how often products are removed from dm+d.

**Reply by rose-higgins** [reply]
There is a somewhat similar issue with deprecated SNOMED codes too, listed in this[ issue here](https://github.com/opensafely-core/opencodelists/issues/1058).

**Comment by Jongmassey** [comment]
In my investigations with OpenSAFELY-TPP, the medication event is not coded directly with the SNOMED/dm+d code, but a Multilex drug id. There is a mapping table in TPP, `MedicationDictionary` which maps this multilex code to the `DMD_ID`. When looking at antibiotic prescriptions in 2019, issues of Amoxicillin under a code deprecated in May 2021 were not found but instead were found under the replacement code, despite its replacement happening after 2019.

https://github.com/opensafely-core/cohort-extractor/blob/76eb09fd43d1bd42133dd537524d2120a0d5ac66/cohortextractor/tpp_backend.py#L1076-L1081

**Comment by wjchulme** [comment]
Just dropping by to say that this fits into a broader class of tools we might want to develop for identifying out-of-date codelists and answering these sorts of questions:

* For a given codelist, have any codes been deprecated? Are there replacements for deprecated codes? In what circumstances should deprecated codes be removed from the codelist?
* For a given codelist created using a term search, are there new codes that appear in the search that didn't exist when the codelist was originally made? What tools can we provide to triage these newly identified codes and incorporate into an updated codelist?
* How should we version codelists that have been updated?
* When should such investigations be run -- user-requested? automatically run when a codelist is downloaded using `opensafely codelist update`? scheduled to run regularly across all codelists? If scheduled, which set of opensafely users should we notify?
